cpp/include/rapidsmpf/streaming/cudf/utils.hpp

```diff
@@ -0,0 +1,39 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <cuda_runtime.h>
+
+#include <rmm/cuda_stream_view.hpp>
+
+#include <rapidsmpf/error.hpp>
+
+namespace rapidsmpf::streaming::utils {
+
+/**
+ * @brief Make @p primary wait until all work currently enqueued on @p secondary
+ * completes.
+ *
+ * Records @p event on @p secondary and inserts a wait for that event on @p primary.
+ * This is fully asynchronous with respect to the host thread; no host-side blocking.
+ *
+ * @param primary The stream that must not run ahead.
+ * @param secondary The stream whose already-enqueued work must complete first.
+ * @param event The CUDA event to use for synchronization. The same event may be reused
+ * across multiple calls; the caller does not need to provide an unique event each time.
+ */
+inline void sync_streams(
```

Comment:
Let's use the new [cuda_stream_join()](https://github.com/rapidsai/rapidsmpf/blob/f93215098f894f1914b380d02dbfc4b4f3708c02/cpp/include/rapidsmpf/cuda_stream.hpp#L33)